vms/shift/tests/test_unit.py

```diff
@@ -410,3 +412,166 @@ def test_model_representation(self):
+        start_time = datetime.time(hour=10, minute=0, second=0)
+        end_time = datetime.time(hour=12, minute=0, second=0)
+        return create_edit_request_with_details(start_time, end_time, logged_shift)
+
+    def test_valid_model_create(self):
+        shift = self.create_shift()
+        volunteer = self.create_volunteer()
+
+        edit_request = self.create_edit_request(volunteer, shift)
+
+        # Check db instance creation
+        self.assertEqual(len(EditRequest.objects.all()), 1)
+
+        shift_in_db = Shift.objects.get(Q(job=self.job))
+        vol_shift_in_db = VolunteerShift.objects.get(Q(shift=shift_in_db))
+        edit_request_in_db = EditRequest.objects.get(Q(volunteer_shift=vol_shift_in_db))
+
+        # verify correctness
+        self.assertEqual(edit_request_in_db.volunteer_shift.volunteer.first_name, edit_request.volunteer_shift.volunteer.first_name)
+        self.assertEqual(edit_request_in_db.volunteer_shift.volunteer.last_name, edit_request.volunteer_shift.volunteer.last_name)
+        self.assertEqual(str(edit_request_in_db.volunteer_shift.shift.start_time), edit_request.volunteer_shift.shift.start_time)
+        self.assertEqual(str(edit_request_in_db.volunteer_shift.shift.end_time), edit_request.volunteer_shift.shift.end_time)
+        self.assertEqual(edit_request_in_db.start_time, edit_request.start_time)
+        self.assertEqual(edit_request_in_db.end_time, edit_request.end_time)
+
+    def test_model_edit_with_valid_values(self):
+        shift = self.create_shift()
+        volunteer = self.create_volunteer()
+
+        edit_request = self.create_edit_request(volunteer, shift)
+
+        # Check db instance creation
+        self.assertEqual(len(EditRequest.objects.all()), 1)
+
+        shift_in_db = Shift.objects.get(Q(job=self.job))
+        volunteer_in_db =  Volunteer.objects.get(Q(first_name='Son'))
+        vol_shift_in_db = VolunteerShift.objects.get(Q(shift=shift_in_db))
+        edit_request_in_db = EditRequest.objects.get(Q(volunteer_shift=vol_shift_in_db))
+
+        # verify correctness
+        self.assertEqual(edit_request_in_db.volunteer_shift.volunteer.first_name, edit_request.volunteer_shift.volunteer.first_name)
+        self.assertEqual(edit_request_in_db.volunteer_shift.volunteer.last_name, edit_request.volunteer_shift.volunteer.last_name)
+        self.assertEqual(str(edit_request_in_db.volunteer_shift.shift.start_time), edit_request.volunteer_shift.shift.start_time)
+        self.assertEqual(str(edit_request_in_db.volunteer_shift.shift.end_time), edit_request.volunteer_shift.shift.end_time)
+        self.assertEqual(edit_request_in_db.start_time, edit_request.start_time)
+        self.assertEqual(edit_request_in_db.end_time, edit_request.end_time)
+
+        volunteer_in_db.first_name = 'Prince'
+        volunteer_in_db.last_name = 'Vegeta'
+        volunteer_in_db.save()
+        vol_shift_in_db.volunteer = volunteer_in_db
+        vol_shift_in_db.save()
```

Comment:
you can do model.save() only one time here after editing all the fields you want to.

Reply:
I am saving two different things @Monal5031